File: androidApp/src/main/java/com/medhelp/medhelp/data/model/BonusesItem.java
package com.medhelp.medhelp.data.model;

import com.google.gson.annotations.SerializedName;
import com.medhelp.medhelp.utils.TimesUtils;

//public class BonusesItem implements Comparable<BonusesItem> {
//    @SerializedName("data_bonus")
//    String date;
//    @SerializedName("sym_bonus")
//    int value;
//    @SerializedName("statys_bonus")
//    String status;
//
//    public String getDate() {
//        return date;
//    }
//
//    public void setDate(String date) {
//        this.date = date;
//    }
//
//    public int getValue() {
//        return value;
//    }
//
//    public void setValue(int value) {
//        this.value = value;
//    }
//
//    public String getStatus() {
//        return status;
//    }
//
//    public void setStatus(String status) {
//        this.status = status;
//    }
//
//    @Override
//    public int compareTo(BonusesItem o) {
//        Long l1= TimesUtils.stringToLong(getDate(),TimesUtils.DATE_FORMAT_ddMMyyyy);
//        Long l2= TimesUtils.stringToLong(o.getDate(),TimesUtils.DATE_FORMAT_ddMMyyyy);
//
//        return l1.compareTo(l2);
//    }
//}

File: shared/src/commonMain/kotlin/com/medhelp/newmedhelp/model/BonusesItem.kt
package com.medhelp.newmedhelp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
open class BonusesItem {
    @SerialName("data_bonus")
    var date: String? = null

    @SerialName("sym_bonus")
    var value = 0

    @SerialName("statys_bonus")
    var status: String? = null

    constructor() {} // for ios

    constructor(item: BonusesItem) {
        date = item.date
        value = item.value
        status = item.status
    }
}